### PR TITLE
Fix cmake install target on windows

### LINKED ISSUE #314: CMake install target does not install dll on windows
Since version 2.3.0 (commit https://github.com/pboettch/json-schema-validator/commit/0d60d48a586965b56d1f5c89a1e7a5cedbbdb8f2) the install target on windows is broken, as predicted by the following comment:

https://github.com/pboettch/json-schema-validator/blob/349cba9f7e3cb423bbc1811bdd9f6770f520b468/src/CMakeLists.txt#L58

The problem seems to be the use of the non-existent `${CMAKE_INSTALL_RUNTIMEDIR}`, replacing it with `${CMAKE_INSTALL_BINDIR}` fixed it for me. The documentation for [install(TARGETS)](https://cmake.org/cmake/help/latest/command/install.html#targets) also suggests `RUNTIME` should be set to `${CMAKE_INSTALL_BINDIR}`.

I've attached a patch, applying this fix.
[fix_cmake.patch](https://github.com/pboettch/json-schema-validator/files/14100814/fix_cmake.patch)

### PR DESCRIPTION
**Pull request by bvstrien**
Fixes #314

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -62,7 +62,7 @@ if (JSON_VALIDATOR_INSTALL)
                 NAMELINK_COMPONENT nlohmann_json_schema_validator_Development
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT nlohmann_json_schema_validator_Development
                 PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nlohmann COMPONENT nlohmann_json_schema_validator_Development
-                RUNTIME DESTINATION ${CMAKE_INSTALL_RUNTIMEDIR} COMPONENT nlohmann_json_schema_validator_Runtime)
+                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT nlohmann_json_schema_validator_Runtime)
```
Comment (LecrisUT) [inline src/CMakeLists.txt:65]: Actually, can you get rid of `WIN32` and merge these two? If `LIBRARY DESTINATION` is not used on `WIN32` and `RUNTIME DESTINATION` is not defined for `UNIX` libraries, these two look identical right? Double-check if I am not dyslexic.

Comment (bvstrien) [reply src/CMakeLists.txt]: Comparing the definitions with each other, the only difference I could find was the added `RUNTIME DESTINATION`.

**Review by LecrisUT — APPROVED** [review]

**Comment by LecrisUT** [comment]
Sorry, can you rebase after #316?

**Comment by LecrisUT** [comment]
Merged that PR, can you rebase?

**Pull request merged**